### PR TITLE
Add default icons for FileAttachment annotations (bug 1230933)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Please note:* This "borrows" the icons from Thunderbird.

According to the PDF specification, see https://web.archive.org/web/20220309040754if_/https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#G11.2096626, we should be providing default icons for FileAttachment annotations without appearances.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
For reference, this is how Adobe Reader handles the referenced PDF document (it doesn't include them when printing, hence we ought to be able to skip that as well):

icons

File: src/display/annotation_layer.js [hunk 1]
```diff
@@ -2509,7 +2509,20 @@ class FileAttachmentAnnotationElement extends AnnotationElement {
   render() {
     this.container.className = "fileAttachmentAnnotation";
 
-    const trigger = document.createElement("div");
+    let trigger;
+    if (this.data.hasAppearance) {
+      trigger = document.createElement("div");
+    } else {
+      // Unfortunately it seems that it's not clearly specified exactly what
```
Comment (calixteman) [inline src/display/annotation_layer.js:2516]: Can we just use the background-image css property and plug it on a new attribute we could add to the div node ?

Comment (Snuffleupagus) [reply src/display/annotation_layer.js:2516]: Unfortunately that doesn't work with the ref-tests (and I don't really have time to try and improve that), and it also doesn't auto-size the icons (meaning they overflow and are clipped) which the `img` approach just handles.

Comment (calixteman) [reply src/display/annotation_layer.js:2516]: Ah yes it's likely the same issue we've with the missing blue background we don't have for some tests with text fields.

**Review by calixteman — APPROVED** [review]
LGTM.

File: src/display/annotation_layer.js (hunk 1 above)
Comment (Snuffleupagus) [reply src/display/annotation_layer.js:2516]: > Ah yes it's likely the same issue we've with the missing blue background we don't have for some tests with text fields.

Aren't those ones actually *intermittent* failures though, even is some of them are happening fairly commonly?

Comment (calixteman) [reply src/display/annotation_layer.js:2516]: My impress is that we've the blue background when we're making ref but not when we're just comparing...
I had a look at some point and if iirc the generated svg is correct but for any reason the image dumper is wrong.
Yes they look like intermittent, but for any reason each time we touch the annotation stuff, I've the impress that they are failing...
Anyway, the blue background is rendered thanks to an image (it's why I made the relation with what you said):
 - https://github.com/mozilla/pdf.js/blob/master/web/annotation_layer_builder.css#L17
 - https://github.com/mozilla/pdf.js/blob/master/web/annotation_layer_builder.css#L91

**Pull request merged**